src/main/java/edu/ucar/unidata/rosetta/controller/TemplateController.java

```diff
@@ -64,29 +50,302 @@
+            return new ModelAndView(new RedirectView("/step2", true));
+        }
+    }
+
+
+    /**
+     * Accepts a GET request for access to step 2 of the wizard.
+     *
+     * @param model  The Model object to be populated.
+     * @param request   HttpServletRequest needed to get the cookie.
+     * @return  View and the Model for the wizard to process.
+     */
+    @RequestMapping(value = "/step2", method = RequestMethod.GET)
+    public ModelAndView displayFileUploadForm(Model model, HttpServletRequest request) {
+
+        // Have we visited this page before during this session?
+        Cookie rosettaCookie = WebUtils.getCookie(request, "rosetta");
+
+        // Create a Data form-backing object.
+        Data data;
+        if (rosettaCookie != null)
+            // Populate Data object with info.
+            data = dataManager.lookupById(rosettaCookie.getValue());
+        else
+            // Something has gone wrong.  We shouldn't be at this step without having persisted data.
+            throw new IllegalStateException("No persisted data available for file upload step.  Check the database & the cookie.");
+
+        // Add data object to Model.
+        model.addAttribute("data", data);
+        // Add current step to the Model.
+        model.addAttribute("currentStep", "2");
+        // Add a list of all steps to the Model for rendering left nav menu.
+        model.addAttribute("steps", resourceManager.loadResources().get("steps"));
+        // Add domains data to Model (for file upload  display based on community type).
+        model.addAttribute("domains", resourceManager.loadResources().get("domains"));
+        // Add file type data to Model (for file type selection if cfType was directly specified).
+        model.addAttribute("fileTypes", resourceManager.loadResources().get("fileTypes"));
+        return new ModelAndView("wizard");
+    }
+
+
+    /**
+     * Accepts a POST request from step 2 of the wizard. Collects the dataFileType and uploaded files
+     * entered by the user and validates them.  If they passes validation, the files are written to the
+     * file system, the dataFileType is written to the database and the user is redirected to step 3.
+     * Otherwise, the user is taken pack to step 2 to correct the invalid data.
+     *
+     * If the user clicked the previous button, they are redirected back to step 1.
+     *
+     * @param data      The form-backing object.
+     * @param result    The BindingResult for error handling.
+     * @param model     The Model object to be populated by file upload data in the next step.
+     * @param request   HttpServletRequest needed to get the cookie.
+     * @return          Redirect to next step.
+     */
+    @RequestMapping(value = "/step2", method = RequestMethod.POST)
+    public ModelAndView processFileUpload(Data data, BindingResult result, Model model, HttpServletRequest request) throws IOException {
```

Comment:
![CRITICAL](https://sonarsource.github.io/sonar-github/severity-critical.png 'Severity: CRITICAL') Refactor this method to reduce its Cognitive Complexity from 21 to the 15 allowed. [![rule](https://sonarsource.github.io/sonar-github/rule.png)](https://sonarcloud.io/coding_rules#rule_key=squid%3AS3776)